src/ansys/mapdl/core/hpc/pyhps.py

```diff
@@ -73,487 +73,846 @@ def get_value_from_json_or_default(
-    project_api,
-    inputs: Optional[list[str]] = None,
-    input_file_id: str = None,
-    outputs: Optional[list[str]] = None,
-    output_file_id: str = None,
-):
-    if inputs is None:
-        inputs = []
-        logger.debug("Setting empty input parameters.")
-
-    def is_float(num):
-        try:
-            float(num)
-            return True
-        except ValueError:
-            return False
-
-    input_params = []
-    param_mappings = []
-
-    for each_input_parm in inputs:
-        name, value = each_input_parm.split("=")
-        if is_float(value):
-            parm = FloatParameterDefinition(
-                name=name,
-                display_text=name,
-                default=value,
+    @memory.setter
+    def memory(self, memory: int):
+        self._memory = self._validate_memory(memory)
+
+    @property
+    def disk_space(self):
+        return self._disk_space
+
+    @disk_space.setter
+    def disk_space(self, disk_space: int):
+        self._disk_space = self._validate_disk_space(disk_space)
+
+    @property
+    def exclusive(self):
+        return self._exclusive
+
+    @exclusive.setter
+    def exclusive(self, exclusive: bool):
+        self._exclusive = self._validate_exclusive(exclusive)
+
+    @property
+    def max_execution_time(self):
+        return self._max_execution_time
+
+    @max_execution_time.setter
+    def max_execution_time(self, max_execution_time: int):
+        self._max_execution_time = self._validate_max_execution_time(max_execution_time)
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, name: str):
+        self._name = self._validate_name(name)
+
+    @property
+    def task_definition(self):
+        return self._task_definition
+
+    @task_definition.setter
+    def task_definition(self, task_definition: TaskDefinition):
+        self._task_definition = task_definition
+
+    def _validate_inputs(self, inputs):
+        """Validate inputs inputs"""
+
+        if self.mode != "python":
+            raise ValueError("Inputs are not supported when using APDL or shell files.")
+
+        if inputs is None:
+            inputs = []
+
+        if isinstance(inputs, str):
+            inputs = inputs.split(",")
+
+        return inputs
+
+    def _validate_outputs(self, outputs):
+        """Validate outputs inputs"""
+
+        if self.mode != "python":
+            raise warn(
+                f"""Outputs are not directly supported when using APDL or shell files.
+                However you can still write the parameters you want to the
+                file {self._output_parms_file} using the python format
+                ('PARAMETER=VALUE')."""
             )
+
+        if outputs is None:
+            outputs = []
+
+        if isinstance(outputs, str):
+            outputs = outputs.split(",")
+
+        return outputs
+
+    def _validate_main_file(self, main_file):
+        """Validate main_file input"""
```

Comment:
```suggestion
        """Validate the main input file."""
```
The brief description should be in plain English.